addons/account/models/account_accrued_entry.py

```diff
@@ -0,0 +1,52 @@
+import json
+from odoo import models, _
+
+
+class AccruedEntryMixin(models.AbstractModel):
+    _name = 'account.accrued.entry.mixin'
+    _description = 'Account Accrued Entry Mixin'
+
+    def _get_preview_data(self, move_vals, currency_id=None):
```

Comment:
```suggestion
    def _get_preview_data(self, record):
```
I think we can just do this (we need to adjust the rest of the code of course).